src/_pytest/_code/code.py

```diff
@@ -818,19 +818,19 @@ def _truncate_recursive_traceback(self, traceback):
 
         return traceback, extraline
 
-    def repr_excinfo(self, excinfo):
-
+    def repr_excinfo(self, excinfo: Optional[ExceptionInfo]) -> "ExceptionChainRepr":
```

Comment:
`Optional` only since it gets re-assigned in the function - did not want to rename it everywhere.. should I do that instead, or is there a better pattern?

Reply:
Yeah that's an annoying situation.

I'd prefer not to change the function type to something incorrect because of an internal detail. So I'd do the rename, but you said you don't like it. The alternative is to set `allow_redefinition = True` mypy option and then you're allowed to add `excinfo = excinfo  # type: Optional[ExceptionInfo]`. It's nice to be able to do that but risk is that it can mask mistakes.

Reply:
ok, will rename it internally then to `excinfo_` or similar (for the loop).

Reply:
Done.